mteb/tasks/PairClassification/multilingual/XStance.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks import MultilingualTask
+from ....abstasks.AbsTaskPairClassification import AbsTaskPairClassification
+
+
+class XStance(MultilingualTask, AbsTaskPairClassification):
+    metadata = TaskMetadata(
+        name="XStance",
+        dataset={
+            "path": "x_stance",
+            "revision": "810604b9ad3aafdc6144597fdaa40f21a6f5f3de",
+        },
+        description="A Multilingual Multi-Target Dataset for Stance Detection in French, German, and Italian.",
+        reference="https://github.com/ZurichNLP/xstance",
+        category="s2s",
+        type="PairClassification",
+        eval_splits=["test"],
+        eval_langs={
+            "de": ["deu-Latn"],
+            "fr": ["fra-Latn"],
+            "it": ["ita-Latn"],
+        },
+        main_score="ap",
+        date=("2011-01-01", "2020-12-31"),
+        form=["written"],
+        domains=["Social"],
+        task_subtypes=["Political classification"],
+        license="cc by-nc 4.0",
+        socioeconomic_status="medium",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="created",
+        bibtex_citation="""
+            @inproceedings{vamvas2020xstance,
+                author    = "Vamvas, Jannis and Sennrich, Rico",
+                title     = "{X-Stance}: A Multilingual Multi-Target Dataset for Stance Detection",
+                booktitle = "Proceedings of the 5th Swiss Text Analytics Conference (SwissText)  16th Conference on Natural Language Processing (KONVENS)",
+                address   = "Zurich, Switzerland",
+                year      = "2020",
+                month     = "jun",
+                url       = "http://ceur-ws.org/Vol-2624/paper9.pdf"
+            }
+        """,
+        n_samples={"test": 2048},
+        avg_character_length={"test": 152.41},  # length of`sent1` + `sent2`
+    )
+
+    def dataset_transform(self):
+        max_n_samples = 2048
+        _dataset = {}
+        for lang in self.hf_subsets:
+            _dataset[lang] = {}
+            for split in self.metadata.eval_splits:
+                # hf config does not support selection by language, use manual filter instead
+                hf_dataset = self.dataset["de"][split].filter(
```

Comment:
I think this should be `self.dataset[lang][split].....` instead of `self.dataset["de"][split]......`

Reply:
The language selection does not work with the HF config. That's why I have the Lang hard coded and do the language filtering manually. It's an issue with the HF dataset implementation. See the comment above.

Reply:
An alternative, for this to look less strange for an outside reader, would be to overwrite the load data (it's often the case, see OpusparcusPC for example) and there load the dataset and assigning the languages properly:
```
def load_data(self, **kwargs):
    """Load dataset from HuggingFace hub"""
    if self.data_loaded:
        return
    self.dataset = {}
    path = self.metadata_dict["dataset"]["path"]
    revision = self.metadata_dict["dataset"]["revision"]
    raw_dataset = datasets.load_dataset(path, revision=revision)
    for split in self.metadata_dict["eval_splits"]:
        self.dataset[split] = {}
        for lang in self.metadata.eval_langs:
            self.dataset[split][lang] = raw_dataset[split].filter(
                lambda row: row["language"] == lang
            )
    self.dataset_transform(lang) # and in your dataset-transform function you can do whatever ops that are missing
    self.data_loaded = True
```